mythril/laser/ethereum/state/account.py

```diff
@@ -91,36 +92,52 @@ def __init__(
         """
         self.nonce = 0
         self.code = code or Disassembly("")
-        self.balance = (
-            balance
-            if balance
-            else symbol_factory.BitVecSym("{}_balance".format(address), 256)
+        self.address = (
+            address
+            if isinstance(address, BitVec)
+            else symbol_factory.BitVecVal(int(address, 16), 256)
         )
+
         self.storage = Storage(
-            concrete_storage, address=address, dynamic_loader=dynamic_loader
+            concrete_storage, address=self.address, dynamic_loader=dynamic_loader
         )
+
         # Metadata
-        self.address = address
         self.contract_name = contract_name
 
         self.deleted = False
 
+        self._balances = balances
+        self.balance = lambda: self._balances[self.address]
+
     def __str__(self) -> str:
         return str(self.as_dict)
 
-    def set_balance(self, balance: ExprRef) -> None:
+    def set_balance(self, balance: Union[int, BitVec]) -> None:
         """
 
         :param balance:
         """
-        self.balance = balance
+        balance = (
+            symbol_factory.BitVecVal(balance, 256)
+            if isinstance(balance, int)
+            else balance
+        )
+        if self._balances is None:
+            assert False
+        self._balances[self.address] = balance
 
-    def add_balance(self, balance: ExprRef) -> None:
+    def add_balance(self, balance: Union[int, BitVec]) -> None:
         """
 
         :param balance:
         """
-        self.balance += balance
+        balance = (
+            symbol_factory.BitVecVal(balance, 256)
+            if isinstance(balance, int)
+            else balance
+        )
+        self._balances[self.address] += balance
 
     @property
     def as_dict(self) -> Dict:
```

Comment:
the parentheses of self.balance are missing here.
it should be `"balance": self.balance()`

Reply:
Good catch